catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1121,6 +1126,44 @@ public CreateResponse create(CreateRequest createRequest)
         return createResponse;
     }
 
+    private MetacardType injectAttributes(MetacardType original) {
+        AttributeRegistry attributeRegistry = frameworkProperties.getAttributeRegistry();
+        InjectableAttributeRegistry injectableAttributeRegistry =
+                frameworkProperties.getInjectableAttributeRegistry();
+        String metacardTypeName = original.getName();
+
+        Set<AttributeDescriptor> injectAttributes =
+                injectableAttributeRegistry.injectableAttributes(metacardTypeName)
+                        .stream()
+                        .map(attributeRegistry::lookup)
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(Collectors.toSet());
+        return new MetacardTypeImpl(original, injectAttributes, metacardTypeName);
+    }
+
+    private Metacard changeMetacardType(Metacard original, MetacardType newMetacardType) {
+        MetacardImpl newMetacard = new MetacardImpl(original);
+        newMetacard.setType(newMetacardType);
+        newMetacard.setSourceId(original.getSourceId());
+        return newMetacard;
+    }
+
+    private CreateRequest injectAttributes(CreateRequest createRequest) {
+        List<Metacard> metacards = createRequest.getMetacards()
+                .stream()
+                .filter(Objects::nonNull)
```

Comment:
Probably, because the caller might want to know that they tried to pass null metacards in their create request. I think if I move the call to `injectAttributes` into the `try` block and remove the `nonNull` filter, everything should still work as expected because we're catching `RuntimeExceptions` inside.